From 9647a01328687e3fdc6cb1714804425c49d6782c Mon Sep 17 00:00:00 2001
From: Shraddha Basantwani <basantwani@google.com>
Date: Mon, 2 Dec 2024 09:36:26 +0000
Subject: [PATCH] Add a list of Biometric protected packages

Protected packages require biometric authentication for modification
(Disable, force-stop or uninstalling updates).

Bug: 352504490, 344865740
Test: atest AppButtonsPreferenceControllerTest PackageInfoPresenterTest

Flag: EXEMPT High Security Bug
DISABLE_TOPIC_PROTECTOR

(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:10e65babc1ca2101508c251e3e39de4cda62f011)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:fc86f8500914dc9a78018aba8d3649c76e67993a)
Merged-In: I0c494e307b02229d751de118abcc89e4e61a6861
Change-Id: I0c494e307b02229d751de118abcc89e4e61a6861
---
 core/res/res/values/config.xml  | 4 ++++
 core/res/res/values/symbols.xml | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 9dcb76040883..cd958e980c48 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -6585,6 +6585,10 @@
          device. -->
     <bool name="config_enableAppCloningBuildingBlocks">true</bool>
 
+    <!-- List of protected packages that require biometric authentication for modification
+         (Disable, force-stop or uninstalling updates). -->
+    <string-array name="config_biometric_protected_package_names" translatable="false" />
+
     <!-- Whether the variable refresh rate when typing feature is enabled for the device. -->
     <bool name="config_variableRefreshRateTypingSupported">false</bool>
 
diff --git a/core/res/res/values/symbols.xml b/core/res/res/values/symbols.xml
index bee5799938f2..74e0977826b2 100644
--- a/core/res/res/values/symbols.xml
+++ b/core/res/res/values/symbols.xml
@@ -5174,6 +5174,9 @@
   <java-symbol type="drawable" name="focus_event_pressed_key_background" />
   <java-symbol type="string" name="lockscreen_too_many_failed_attempts_countdown" />
 
+  <!-- List of protected packages that require biometric authentication for modification -->
+  <java-symbol type="array" name="config_biometric_protected_package_names" />
+
   <!-- Whether we order unlocking and waking -->
   <java-symbol type="bool" name="config_orderUnlockAndWake" />
 </resources>
-- 
2.34.1

